src/Chainweb/Rosetta/Internal.hs

```diff
@@ -264,11 +301,15 @@ gasTransactionAcc accTyp txa@(TxAccumulator logs' _) ctx = combine logs'
     combine (f:[]) = listErr "Only fund logs found" f
     combine [] = listErr "No logs found" ([] :: [(TxId, [AccountLog])])
 
-    makeAcc restLogs fund transfer gas = pure $
-      accumulatorFunction accTyp txa restLogs tx
+    makeAcc restLogs fund transfer gas = pure $!
+      accumulatorFunction accTyp acc restLogs tx
       where
-        tx = makeRosettaTx ctx $ indexedOperations $
-             fund <> transfer <> gas
+        tx = makeRosettaTx ctx cid $! indexedOperations $!
```

Comment:
All the changes that add `ChainId` is because its consumed here by `makeRosettaTx` to give more insight into related transactions.